src/org/daisy/dotify/formatter/impl/obfl/OBFLExpressionBase.java

```diff
@@ -129,6 +135,9 @@ public OBFLExpressionBase(String exp, ExpressionFactory ef, OBFLVariable... vari
         if (metaPageNumberVariableName != null) {
             variables.put(metaPageNumberVariableName, "" + context.getMetaPage());
         }
+        if (startsAtTopOfPageName != null && context instanceof BlockContext) {
```

Comment:
You may consider throwing an exception if `startsAtTopOfPageName != null` and `!(context instanceof BlockContext)`
In this way, the error is ignored silently.

Reply:
Changed code and will throw an IllegalArgument exception if the incorrect context is used.